### DISCUSSION TITLE
Support boundingRect(forGlyphRange)

### DISCUSSION THREAD

**Discussion by aaronvegh** [krzyzanowskim/STTextView#16] [Questions]
I'm building an app that relies extensively on NSTextView. I've been finding of late that drawing performance (particularly when scrolling and resizing these views) left much to be desired. I was hoping that STTextView could be a good replacement!

Unfortunately, one feature that I'm using, that doesn't seem to be available here, is `NSLayoutManager`'s `boundingRect(forGlyphRange)` method. I did some spelunking and it seems `NSLayoutManager` is largely not being used in this project. I can imagine that this request might be a heavy lift.

As for my use case, my application shows an `NSView` above particular blocks of text. Here's an example:

<img width="531" alt="Screenshot 2023-03-27 at 10 08 17 AM" src="https://user-images.githubusercontent.com/34525/227964037-73ced906-f893-4f61-8cd9-7311154affd9.png">

I derive the bounds of the view by calling that layout manager function to ensure that the view perfectly covers the range of the text at issue.

Let me know if you have any questions! Thank you!

**Comment by krzyzanowskim** [comment]
STTextView does not use `NSLayoutManager` (TextKit1) because it uses new `NSTextLayoutManager` from TextKit2. I think you can use it to get the bounds for the range. The closest you need is probably the helper I use for selection:

`NSTextLayoutManager.textSelectionSegmentFrame(in:type:)`

https://github.com/krzyzanowskim/STTextView/blob/81eeabedc2cb58cf1ccf7abed2307aec1dd4bb34/Sources/STTextView/Extensions/NSTextLayoutManager%2BHelpers.swift#L44

I use it for annotations in the Demo project https://github.com/krzyzanowskim/STTextView/blob/81eeabedc2cb58cf1ccf7abed2307aec1dd4bb34/STTextViewDemo/ViewController.swift#L115

let me know if you know how to use it of if you need more help on that?
